packages/core/src/lib/pagination/layout.type.mts

```diff
@@ -0,0 +1,6 @@
+/** Represents an item that's part of a layout, about to be rendered */
+export type Layout<T> = {
+  item: T;
+  index: number;
```

Comment:
We only use `index` in the test. Is it important for the testing?

If we don't have a clear use for that extra argument, let's remove it to keep the exposed interface lean.

Reply:
Consumers of renderItem might want to render the index of the item as well - it becomes impossible to do that without this argument.

Reply:
We could also just take a count/size and just use (index, isActive) in renderItem - then it becomes the consuming component's responsibility to lookup the index in its own list of items - then we no longer work on items of type `T`. What do you think?

Reply:
That also means renderItem can only be defined with a closure to items, and not without it.

Reply:
I'm okay with how it is now.